flatpak_builder_lint/checks/desktop.py

```diff
@@ -116,6 +116,19 @@ def _validate(self, path: str, appid: str) -> None:
             except KeyError:
                 pass
 
+            try:
+                # https://github.com/ximion/appstream/blob/146099484012397f166cd428c56f230487b2d1fc/src/as-desktop-entry.c#L144-L154
+                # appstreamcli filters these out during compose, "GUI"
+                # is caught by desktop-file-validate
+                block = {"KDE", "GTK", "Qt", "Application", "GNOME"}
+                catgr: Set[str] = set(
+                    filter(None, d["Desktop Entry"]["Categories"].split(";"))
+                )
+                if len(catgr) > 0 and catgr.issubset(block):
+                    self.warnings.add("desktop-file-low-quality-category")
```

Comment:
Should this be a warning or an error? I kept it warning to stay safe.

Reply:
Let's stick to a warning until we settle on a sustainable way of communicating these.